src/jobservice/period/enqueuer.go

```diff
@@ -91,19 +93,34 @@ func (pe *periodicEnqueuer) loop() {
 				if err != nil {
 					logger.Errorf("periodic_enqueuer.loop.enqueue:%s\n", err)
 				}
+			} else {
+				logger.Debug("Enqueue condition not matched, do nothing.")
 			}
 		}
 	}
 }
 
 func (pe *periodicEnqueuer) enqueue() error {
 	now := time.Now().Unix()
-	nowTime := time.Unix(now, 0)
-	horizon := nowTime.Add(periodicEnqueuerHorizon)
 
 	conn := pe.pool.Get()
 	defer conn.Close()
 
+	// Set last periodic enqueue timestamp in advance to avoid duplicated enqueue actions
+	if _, err := conn.Do("SET", utils.RedisKeyLastPeriodicEnqueue(pe.namespace), now); err != nil {
+		return err
+	}
+
+	// Avoid schedule in the same time slot.
+	expireTime := (int64)(periodicEnqueuerSleep/time.Minute) * (60 + 5)
+	slot := fmt.Sprintf("%s:%d", utils.RedisKeyLastPeriodicEnqueue(pe.namespace), now)
```

Comment:
1) This chunk is repetitive with line 110~113.
2) `fmt.Sprintf("%s:%d", utils.RedisKeyLastPeriodicEnqueue(pe.namespace), now)` this may not work in the multi-instance scenario, as the value of `now` in different instances may be different.